src/ledger/LedgerTxnAccountSQL.cpp

```diff
@@ -330,4 +330,70 @@ LedgerTxnRoot::Impl::dropAccounts()
         << "CREATE INDEX accountbalances ON accounts (balance) WHERE "
            "balance >= 1000000000";
 }
+
+static std::unordered_map<std::string, std::string>
+loadHomeDomainsToEncode(Database& db, size_t limit, size_t offset)
+{
+    std::string accountID, homeDomain;
+
+    std::string sql = "SELECT accountid, homedomain FROM accounts "
+                      "ORDER BY accountid LIMIT :v1 OFFSET :v2";
+
+    auto prep = db.getPreparedStatement(sql);
+    auto& st = prep.statement();
+    st.exchange(soci::into(accountID));
+    st.exchange(soci::into(homeDomain));
+    st.exchange(soci::use(limit));
+    st.exchange(soci::use(offset));
+    st.define_and_bind();
+    st.execute(true);
+
+    std::unordered_map<std::string, std::string> res;
+    while (st.got_data())
+    {
+        res.emplace(accountID, homeDomain);
+        st.fetch();
+    }
+    return res;
+}
+
+static void
+writeEncodedHomeDomain(Database& db, std::string const& accountID,
+                       std::string const& homeDomain)
+{
+    std::string encodedHomeDomain = decoder::encode_b64(homeDomain);
+
+    std::string sql =
+        "UPDATE accounts SET homedomain = :v1 WHERE accountid = :v2";
+
+    auto prep = db.getPreparedStatement(sql);
+    auto& st = prep.statement();
+    st.exchange(soci::use(encodedHomeDomain));
+    st.exchange(soci::use(accountID));
+    st.define_and_bind();
+    st.execute(true);
+}
+
+void
+LedgerTxnRoot::Impl::encodeHomeDomainsBase64()
+{
+    throwIfChild();
+    mEntryCache.clear();
+    mBestOffersCache.clear();
+
+    size_t const LIMIT = 4096;
+    size_t offset = 0;
+    auto homeDomainsToEncode =
```

Comment:
same thing, you can safely ditch the LIMIT/offset and related logic, with the filtering we're talking less than 100k accounts which is safe to keep in memory